x-pack/plugins/gis/public/actions/store_actions.js

```diff
@@ -296,20 +296,22 @@ export function updateSourceProp(layerId, propName, value) {
       propName,
       value,
     });
-
     dispatch(syncDataForLayer(layerId));
   };
 }
 
 export function syncDataForLayer(layerId) {
-  return (dispatch, getState) => {
+  return async (dispatch, getState) => {
```

Comment:
Why should dispatching this state wait for `syncData`. This is not a null op for VectorLayer. VectorLayer.syncData has some awaits. I think the state change should just fire. No need to wait for anything (unless there is and then lets add a comment about what we are waiting for and why)

Reply:
because otherwise you can't use these actions and chain them one after the other, which is a fairly common pattern.

If you would call the `syncDataForLayer` action and wait for its completion (so you can run other actions behind it), imho it's a fair assumption that that method shouldn't return before having fully completed. (right now, it would return before `Layer#syncData` has resolved, which doesn't seem right).